data api/src/main/java/de/tinf15b4/kino/api/controllers/SecureRestController.java

```diff
@@ -179,6 +179,46 @@
         return ResponseEntity.ok(RestControllerConstants.DELETE_SUCCESSFUL);
     }
 
+    @ApiMethod(description = "Saves the given user for the given token")
+    @ApiErrors(apierrors = { @ApiError(code = "401", description = "Token was invalid"),
+            @ApiError(code = "406", description = "user was null or invalid") })
+    @RequestMapping(value = "rest/saveUser", method = RequestMethod.POST)
+    public ResponseEntity<?> saveUser(
+            @ApiQueryParam(name = "token", description = "Authentication token for the current user") @RequestParam(name = "token") String token,
+            @ApiBodyObject(clazz = User.class) @RequestBody User user) {
+        ResponseEntity<?> response = checkToken(token);
+        if (response.getStatusCode() != HttpStatus.OK)
+            return response;
+        if (user == null)
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(RestControllerConstants.NOT_NULL);
+        if (user.getPassword() == null || user.getPassword().length() < 7 || user.getPassword().length() > 99)
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(RestControllerConstants.INVALID_USERDATA);
+        if (user.getName() == null || user.getName().length() < 3 || user.getName().length() > 99)
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(RestControllerConstants.INVALID_USERDATA);
+        if (user.getEmail() == null ||
+                user.getEmail().length() < 5 ||
+                !user.getEmail().contains(".") ||
+                !user.getEmail().contains("@") ||
+                user.getEmail().length() > 99)
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(RestControllerConstants.INVALID_USERDATA);
+        Optional<User> updated = userService.save(user);
+        Token userToken = new Token(token);
+        String oldUserName = tokens.get(userToken);
+
+        if (oldUserName != null){
+            tokens.remove(userToken);
+            tokens.put(userToken, user.getName());
+            users.remove(oldUserName);
+            users.put(user.getName(), userToken);
+        }
+
+        if (updated.isPresent()) {
```

Comment:
This check should probably be done before we update the token maps. In case something goes wrong with saving updating the maps might cause trouble